src/PayPal/Order/CommandHandler/CapturePayPalOrderCommandHandler.php

```diff
@@ -66,35 +68,82 @@ class CapturePayPalOrderCommandHandler
      * @var PayPalCustomerRepository
      */
     private $payPalCustomerRepository;
+
+    /** @var PayPalConfiguration  */
+    private $payPalConfiguration;
+
+    /** @var PayPalOrderRepository */
+    private $payPalOrderRepository;
     /**
-     * @var PayPalOrderRepository
+     * @var LoggerInterface
      */
-    private $payPalOrderRepository;
+    private $logger;
 
     public function __construct(
-        MaaslandHttpClient $maaslandHttpClient,
+        MaaslandHttpClient       $maaslandHttpClient,
         EventDispatcherInterface $eventDispatcher,
-        CacheInterface $orderPayPalCache,
-        PrestaShopContext $prestaShopContext,
+        CacheInterface           $orderPayPalCache,
+        PrestaShopContext        $prestaShopContext,
         PayPalCustomerRepository $payPalCustomerRepository,
-        PayPalOrderRepository $payPalOrderRepository
+        LoggerInterface          $logger,
+        PayPalConfiguration      $payPalConfiguration,
+        PayPalOrderRepository    $payPalOrderRepository
     ) {
         $this->maaslandHttpClient = $maaslandHttpClient;
         $this->eventDispatcher = $eventDispatcher;
         $this->orderPayPalCache = $orderPayPalCache;
         $this->prestaShopContext = $prestaShopContext;
         $this->payPalCustomerRepository = $payPalCustomerRepository;
+        $this->payPalConfiguration = $payPalConfiguration;
         $this->payPalOrderRepository = $payPalOrderRepository;
+        $this->logger = $logger;
     }
 
     public function handle(CapturePayPalOrderCommand $capturePayPalOrderCommand)
     {
-        $merchantId = Configuration::get('PS_CHECKOUT_PAYPAL_ID_MERCHANT', null, null, $this->prestaShopContext->getShopId());
+        $merchantId = $this->payPalConfiguration->getMerchantId();
+
+        $capturePayload = $this->buildCapturePayload($capturePayPalOrderCommand);
+
+        $orderPayPal = $this->captureOrder($capturePayload);
+
+        if (empty($orderPayPal)) {
+            throw new \Exception('Order PayPal not found');
+        }
+
+        if (isset($orderPayPal['payment_source'][$capturePayPalOrderCommand->getFundingSource()]['attributes']['vault'])) {
+            $vault = $orderPayPal['payment_source'][$capturePayPalOrderCommand->getFundingSource()]['attributes']['vault'];
+            $vault = $this->savePrestaShopPayPalCustomerRelationship($vault);
```

Comment:
To me this part of overwriting $vault variable feels a bit off. Why not wrap it in try-catch?